### ISSUE TITLE
Assembly indefinitly extending toward the the end

### ISSUE THREAD

**Issue by ClemSc** [ndierckx/NOVOPlasty#124]
Hello,

I am trying to use NOVOPlasty to assemble an hexapod mitogenome from 150 bp paired reads. Contig is fine until a point where a 79bp sequence is indefinitely repeated at the end of the contig, the assembly stops when reach the genome range upper limit. Genome is only partial (12 kb when it should be slightly above 15 kb). Do you know of this kind of behavior ? I append the log file content below

Best  !
Clem

-----------------------------------------------
NOVOPlasty: The Organelle Assembler
Version 3.8.2
Author: Nicolas Dierckxsens, (c) 2015-2019
-----------------------------------------------

Input parameters from the configuration file:   *** Verify if everything is correct ***

Project:
----------------------
Project name          = P4_20
Type                  = mito
Genome range          = 11000-16000
K-mer                 = 32
Max memory            = 50
Extended log          = 0
Save assembled reads  = no
Seed Input            = ../../../Seeds/Seed.fa
Extend seed directly  = no
Reference sequence    = 
Variance detection    = 
Chloroplast sequence  = 

Dataset 1:
----------------------
Read Length           = 151
Insert size           = 300
Platform              = illumina
Single/Paired         = PE
Combined reads        = 
Forward reads         = ../../../Raw_Clipped/P4_20_1.paired.fq
Reverse reads         = ../../../Raw_Clipped/P4_20_2.paired.fq

Heteroplasmy:
-----------------------
Heteroplasmy          = 
HP exclude list       = 
PCR-free              = 

Optional:
----------------------
Insert size auto      = yes
Insert range          = 1.8
Insert range strict   = 1.3
Use Quality Scores    = 

Subsampled fraction: 99.99 %
Forward reads without pair: 933291
Reverse reads without pair: 927455

Retrieve Seed...

Initial read retrieved successfully: ACTACTCTTGTCTTTACCAGTACTAGCAGGTGCTATTACTATGCTTCTTACGGATCGTAATTTGAATACTACATTTTTTGACCCTGCAGGAGGGGGAGACCCTATTCTATATCAACACCTATTCTGATTCTTTGGGCACCCTGAAGTTTA

Start Assembly...

------------Assembly 1 finished: Contigs are automatically merged in Merged_contigs file------------

Contig 01                  : 16025 bp

Total contigs              : 1
Largest contig             : 16025 bp
Smallest contig            : 16025 bp
Average insert size        : 267 bp

-----------------------------------------Input data metrics-----------------------------------------

Total reads                : 64477860
Aligned reads              : 89260
Assembled reads            : 71406
Average organelle coverage : 841

**Comment by ndierckx** [comment]
Hi,

Could you run it again with extended log to 1 and send me that file?

**Comment by ClemSc** [comment]
Done.
Also I used the produced contig as a seed for another de novo assembly pipeline (GetOrganelle), this tool was able to extend the genome to ~15 kb and seems nearly complete (but nor circular). It could not extend from the problematic end (it actually added a few bases of the repeated sequence and then stopped). Maybe this means that there is a complex sequence there that cannot be resolved with short reads...

[log_extended_P4_20.txt](https://github.com/ndierckx/NOVOPlasty/files/4198983/log_extended_P4_20.txt)

**Comment by ndierckx** [comment]
Yeah, it's definitely a long repetitive sequence and it must be very long because the coverage in that region goes from around 300 to 30000, so that repeat sequence could have a count of around 100. This sequence is around 150 bp long so it could be that your genome is longer than 16 kb. If the repeat was shorter, NOVOPlasty can jump over that region by looking for the flanking regions, but this one seems too long.. I could manually change the parameters so it can still do it but you will never know the exact count of repeats with short reads... Did that 15 kb assembly end in that repetitive sequence at both sides? if so I think it's the best you will get. If you send that sequence I can have a look if you want.

**Comment by ClemSc** [comment]
Yes, I am figuring this out with the assembly graph from the other tool, my genome seems to be complete except for this highly repeated region. Assembly can simply not be done in this part with short reads it seems, but the length could roughly be inferred with coverage.

Therefore I would simply point out that this case is a possible stumbling block for Novoplasty.
Thank you for the hints !
